src/Engine/Manager/PhysicsManager.cpp

```diff
@@ -143,30 +159,104 @@ Component::Physics* PhysicsManager::CreatePhysics(const Json::Value& node) {
+        rigidBodyComp->GetBulletRigidBody()->setCollisionShape(shapeComp->GetShape().GetShape());
+    }
+
+    return physics;
+}
+
+Component::RigidBody* PhysicsManager::CreateRigidBody(Entity* owner) {
+    auto comp = rigidBodyComponents.Create();
+    comp->entity = owner;
+
+    comp->NewBulletRigidBody(1.0f);
+
+    auto shapeComp = comp->entity->GetComponent<Component::Shape>();
+    if (shapeComp) {
+        comp->GetBulletRigidBody()->setCollisionShape(shapeComp->GetShape().GetShape());
+    }
+
+    return comp;
+}
+
+Component::RigidBody* PhysicsManager::CreateRigidBody(Entity* owner, const Json::Value& node) {
+    auto comp = rigidBodyComponents.Create();
+    comp->entity = owner;
+
+    auto mass = node.get("mass", 1.0f).asFloat();
+    comp->NewBulletRigidBody(mass);
+
+    auto shapeComp = comp->entity->GetComponent<Component::Shape>();
+    if (shapeComp) {
+        comp->GetBulletRigidBody()->setCollisionShape(shapeComp->GetShape().GetShape());
+    }
+
+    return comp;
+}
+
+Component::Shape* PhysicsManager::CreateShape(Entity* owner) {
+    auto comp = shapeComponents.Create();
+    comp->entity = owner;
+
+    auto rigidBodyComp = comp->entity->GetComponent<Component::RigidBody>();
+    if (rigidBodyComp) {
+        rigidBodyComp->GetBulletRigidBody()->setCollisionShape(comp->GetShape().GetShape());
+    }
+
+    return comp;
+}
+
+Component::Shape* PhysicsManager::CreateShape(Entity* owner, const Json::Value& node) {
+    auto comp = shapeComponents.Create();
+    comp->entity = owner;
+
+    if (node.isMember("sphere")) {
+        auto sphere = node.get("sphere", {});
         auto radius = sphere.get("radius", 1.0f).asFloat();
         auto shape = new ::Physics::Shape(::Physics::Shape::Sphere(radius));
-        physics->rigidBody = new ::Physics::RigidBody(shape, 1.0f);
-    } else if (shape.isMember("plane")) {
-        auto plane = shape.get("plane", {});
+        comp->SetShape(shape);
+    }
+    else if (node.isMember("plane")) {
+        auto plane = node.get("plane", {});
         auto normal = Json::LoadVec3(plane.get("normal", {}));
         auto planeCoeff = plane.get("planeCoeff", 0.0f).asFloat();
         auto shape = new ::Physics::Shape(::Physics::Shape::Plane(normal, planeCoeff));
-        physics->rigidBody = new ::Physics::RigidBody(shape, 1.0f);
+        comp->SetShape(shape);
     }
-    
-    assert(physics->rigidBody);
-    
-    return physics;
+
+    auto rigidBodyComp = comp->entity->GetComponent<Component::RigidBody>();
+    if (rigidBodyComp) {
+        rigidBodyComp->GetBulletRigidBody()->setCollisionShape(comp->GetShape().GetShape());
+    }
+
+    return comp;
+}
+
+void PhysicsManager::SetShape(Component::Shape* comp, ::Physics::Shape* shape) {
+    comp->SetShape(shape);
+}
+
+void PhysicsManager::SetMass(Component::RigidBody* comp, float mass) {
+    comp->Mass(mass);
 }
 
 const std::vector<Component::Physics*>& PhysicsManager::GetPhysicsComponents() const {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The function 'GetPhysicsComponents' is never used. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)